### PR TITLE
[#257] Add jsoncons package

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Addresses #257

Still testing

### CODE REVIEW

File: versions.json [hunk 1]
```diff
@@ -823,6 +823,24 @@
         ],
         "fpm_directories": ["include","lib"]
     },
+    "jsoncons": {
+        "git_repository": "https://github.com/danielaparker/jsoncons.git",
```
Comment (korydraughn) [inline versions.json:827]: NTS: Remember to fork this repo into the iRODS namespace once this PR is ready for merging.

File: versions.json [hunk 2]
```diff
@@ -823,6 +823,24 @@
...
+        "commitish": "v0.178.0",
+        "version_string": "0.178.0",
+        "license": "BSL-1.0",
+        "consortium_build_number": "0",
+        "package_revision": "0",
+        "externals_root": "opt/irods-externals",
+        "build_steps": [
+            "mkdir -p build",
+            "cd build; TEMPLATE_CMAKE_EXECUTABLE .. -G'Unix Makefiles' -DCMAKE_USER_MAKE_RULES_OVERRIDE=TEMPLATE_SCRIPT_PATH/ClangOverrides.txt -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=TEMPLATE_INSTALL_PREFIX -DCMAKE_INSTALL_LIBDIR=lib -DJSONCONS_BUILD_TESTS=OFF",
+            "cd build; make -jTEMPLATE_JOBS; make install"
```
Comment (korydraughn) [inline versions.json:837]: What does the compiled version give us?

The README doesn't appear to say anything about that. Or perhaps this about capturing the directory structures produced by `make install`.

Comment (SwooshyCueb) [reply versions.json:837]: There's no compilation, actually. In CMake it's a header-only library. The build steps here are for generating the CMake package configuration.

Comment (korydraughn) [reply versions.json:837]: Got it.

**Comment by SwooshyCueb** [comment]
Package built successfully for every distro we support, and nothing seems out of the ordinary. Marking ready for review.

File: versions.json (hunk 1 above)
Comment (trel) [reply versions.json:827]: forked it.    can easily sync it back up if it gets to 1.0.0 soonish...

Comment (trel) [reply versions.json:827]: https://github.com/irods/jsoncons

**Comment by trel** [comment]
and are we .... waiting for 1.0.0 to get tagged/released?

File: versions.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (korydraughn) [reply versions.json:827]: The property just needs to be removed now.

**Comment by korydraughn** [comment]
No need for 1.0 at the moment. We only use the validation component of the lib. If 1.0 is released before iRODS 5, then we can bump the jsoncons version.

File: versions.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (korydraughn) [reply versions.json:827]: Once `git_repository` is removed, please verify it can still be built.

[the remark this answers is not in the document]
Comment (SwooshyCueb) [reply versions.json]: Still builds fine

**Review by korydraughn — APPROVED** [review]
Pound it.

If you have the package for each OS, please set those aside so we can get them to @trel for publishing.

**Review by alanking — APPROVED** [review]

**Comment by trel** [comment]
agreed - let's do them all at once please - whenever is good.

**Pull request merged**